gitmanager.py

```diff
@@ -306,20 +306,56 @@ def remove_from_blacklist(cls, item, username, blacklist_type="", code_privilege
     @classmethod
     def merge_pull_request(cls, pr_id, comment=""):
         response = requests.get("https://api.github.com/repos/{}/pulls/{}".format(GlobalVars.bot_repo_slug, pr_id))
-        if not response:
+        files_list = requests.get("https://api.github.com/repos/{}/pulls/{}/files"
+                                  .format(GlobalVars.bot_repo_slug, pr_id))
+        if not response or not files_list:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
+        files_info = files_list.json()
         if pr_info["user"]["login"] != "SmokeDetector":
             raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
         if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
             raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        pr_string = pr_info["title"]
+        file_modified = files_info[0]["filename"]
         ref = pr_info['head']['ref']
+        requested_pr_string = regex.match(r".*?: \S+ (.*?)(?:\Z|\s*\(\?#)", pr_string).group(1)
+        username_of_pr_requester = regex.match(r".*(?=:)", pr_string).group(0)
```

Comment:
This will get the wrong information if there's a `:` within the regex which is watched ([example](https://regex101.com/r/GkWMfO/1/). If you're going to use a regex here, then `regex.match(r"^.*?(?=: )"` is probably reasonable ([example](https://regex101.com/r/B0Zocz/1)).

Reply:
You mentioned a few issues with the RegEx which said you'll work on around 2 months ago, I assume this was one of them...

Reply:
I have no idea how anything which I've said I'll work on affects this specific issue. If you think it does and will remove the concern I've raised here, please elaborate.